syft/pkg/cataloger/cpp/parse_conanlock.go

```diff
@@ -0,0 +1,52 @@
+package cpp
+
+import (
+	"encoding/json"
+	"io"
+	"strings"
+
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/common"
+)
+
+// integrity check
+var _ common.ParserFn = parseConanlock
+
+// parseConanlock is a parser function for conan.lock contents, returning all packages discovered.
+func parseConanlock(_ string, reader io.Reader) ([]*pkg.Package, []artifact.Relationship, error) {
+	pkgs := []*pkg.Package{}
+	var graphLock struct {
+		GraphLock struct {
+			Nodes map[string]struct {
+				Ref string
+			}
+		} `json:"graph_lock"`
+	}
```

Comment:
It might be nice to define this struct at the top level of this file and also be explicit about each field using `json:""` tags.